### PR TITLE
change(scan): Create a function that scans one block by height, and stores the results in the database

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

This is a quick follow up to a bunch of recent scanner changes.

@upbqdn or @oxarbitrage this might conflict with what you're working on, but I'm happy to re-do it, because your PRs are actually changing things.

#### Check before marking the PR as ready for review:
  - [x] Will the PR name make sense to users?
  - [x] Does the PR have a priority label?
  - [x] Have you added or updated tests?
  - [x] Is the documentation up to date?

_If a checkbox isn't relevant to the PR, mark it as done._

## Solution

- Create a function that scans one block by height, gets it from the state, and puts the results in the database
- Add some TODOs and update some function comments

Most of these arguments are cheap to clone, but if we have a lot of keys the key sets could be very large, so I used `Arc`.

### Testing

This is just a cleanup of the previous PR.

## Review

This is a routine cleanup.

### Reviewer Checklist

Check before approving the PR:
  - [ ] Does the PR scope match the ticket?
  - [ ] Are there enough tests to make sure it works? Do the tests cover the PR motivation?
  - [ ] Are all the PR blockers dealt with?
        PR blockers can be dealt with in new tickets or PRs.

_And check the PR Author checklist is complete._

### CODE REVIEW

File: zebra-scan/src/scan.rs [hunk 1]
```diff
@@ -80,115 +81,159 @@ pub async fn start(
...
+            .expect("a valid blockchain never reaches the max height");
+    }
+}
+
+/// Get the block at `height` from `state`, scan it with the keys in `parsed_keys`, and store the
+/// results in `storage`. If `height` is lower than the `key_birthdays` for that key, skip it.
+///
+/// Returns:
+/// - `Ok(Some(height))` if the height was scanned,
+/// - `Ok(None)` if the height was not in the database, and
```
Comment (upbqdn) [inline zebra-scan/src/scan.rs]:
```suggestion
/// - `Ok(None)` if the height was not in the state, and
```

**Review by upbqdn — DISMISSED** [review]

File: zebra-scan/src/scan.rs (hunk 1 above)
Comment (teor2345) [reply zebra-scan/src/scan.rs]: Fair point, since we have multiple databases now.

**Review by upbqdn — DISMISSED** [review]

**Comment by teor2345** [comment]
Just an import merge conflict, should be fixed now.

File: zebra-scan/src/scan.rs [hunk 2]
```diff
@@ -22,8 +22,12 @@ use zcash_primitives::{
...
 
 use zebra_chain::{
-    block::Block, chain_tip::ChainTip, diagnostic::task::WaitForPanics, parameters::Network,
-    serialization::ZcashSerialize, transaction::Transaction,
+    block::{Block, Height},
+    chain_tip::ChainTip,
+    diagnostic::task::WaitForPanics,
+    parameters::Network,
+    serialization::ZcashSerialize,
+    transaction::{self, Transaction},
```
Comment (teor2345) [inline zebra-scan/src/scan.rs]:
```suggestion
    transaction::Transaction,
```

**Review by upbqdn — APPROVED** [review]

**Pull request merged**